documenting the use of the side drawer (#242)
nativescript-vue/nativescript-vue.org: content/docs/en/elements/components/side-drawer.md

---
title: SideDrawer

> apiRef: https://docs.telerik.com/devtools/nativescript-ui/Controls/NativeScript/SideDrawer/overview

Reviewer: Probably link should be: https://docs.telerik.com/devtools/nativescript-ui/Controls/Vue/SideDrawer/overview or the actual API reference at http://docs.telerik.com/devtools/nativescript-ui/api/classes/radsidedrawer.html

> contributors: [MisterBrownRSA, rigor789, eddyverbruggen, ikoevska, sombriks]

Reviewer: Remove all the contributors except for yourself.

---

> `<RadSideDrawer>` is a UI side drawer component which plays a navigation role on application structure.

Reviewer: ```suggestion
`<RadSideDrawer>` is a UI component that provides side drawer navigation for apps.
```

> ## Installing the side drawer:

Reviewer: ```suggestion
## Installing the side drawer
```

> 

Reviewer: `<RadSideDrawer>` is not available with the core NativeScript installation. You need to add it as a plugin to your project.

```shell
$ tns plugin add nativescript-ui-sidedrawer
```

> After that, you need to register it as a vue component:

Reviewer: Specify where this code needs to be added. Is it at the top of some file or code block in a file, which file?

No colon at the end of the sentence.

```javascript
import RadSideDrawer from "nativescript-ui-sidedrawer/vue";
Vue.use(RadSideDrawer);
```

> Then you can use it on your .vue file:

Reviewer: ```suggestion
You can now use the component on your `.vue` file.
```

> 

Reviewer: ```suggestion
## Basic use
```

```html
<Page>
  <ActionBar title="Hello side drawer!">
    <ActionItem text="Open" @tap="$refs.drawer.nativeView.showDrawer()"/>
  </ActionBar>
  <GridLayout rows="*">
    <RadSideDrawer ref="drawer">
      <StackLayout ~drawerContent>
        <StackLayout height="56" style="text-align: center; vertical-align: center;">
          <Label text="Navigation Menu"/>
        </StackLayout>
        <StackLayout>
          <Button text="Friends" @tap="$navigateTo('Friends')"/>
          <Button text="Posts" @tap="$navigateTo('Posts')"/>
        </StackLayout>
      </StackLayout>
      <StackLayout ~mainContent>
        <Label text="This is the main content for the current Page!"
          textWrap="true" fontSize="13" padding="10"/>
      </StackLayout>
    </RadSideDrawer>
  </GridLayout>
</Page>
```

## Props

| Name          | Type | Description                                              |
| ------------- | ---- | -------------------------------------------------------- |

> | drawerContent | View | the UI element containing the drawer content             |

Reviewer: ```suggestion
| `drawerContent` | `View` | The visual elements displayed in the side drawer. |
```

> | mainContent   | View | the UI element being displayed when the drawer is closed |

Reviewer: ```suggestion
| `mainContent`   | `View` | The visual elements displayed in the host view where the drawer is shown. |
```
